hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/util/Shell.java

```diff
@@ -1442,7 +1441,7 @@ public static void destroyAllShellProcesses() {
    */
   public static Set<Shell> getAllShells() {
     synchronized (CHILD_SHELLS) {
-      return new HashSet<>(CHILD_SHELLS.keySet());
+      return CHILD_SHELLS.keySet();
```

Comment:
I think the diff changes the behavior. Would you revert the change?
- Before: The method returns `HashSet`
- After: The method returns `SynchronizedSet` because `CHILD_SHELLS` is `SynchronizedMap`.

Reply:
reverted the changes !